pkg/controller/groups/groupmembers/controller.go

```diff
@@ -0,0 +1,193 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package groupmembers
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/xanzy/go-gitlab"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane-contrib/provider-gitlab/apis/groups/v1alpha1"
+	"github.com/crossplane-contrib/provider-gitlab/pkg/clients"
+	"github.com/crossplane-contrib/provider-gitlab/pkg/clients/groups"
+)
+
+const (
+	errNotGroupMember = "managed resource is not a Gitlab Group Member custom resource"
+	errCreateFailed   = "cannot create Gitlab Group Member"
+	errUpdateFailed   = "cannot update Gitlab Group Member"
+	errDeleteFailed   = "cannot delete Gitlab Group Member"
+)
+
+// SetupGroupMember adds a controller that reconciles Group Members.
+func SetupGroupMember(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.GroupMemberKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.GroupMember{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.GroupMemberKubernetesGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newGitlabClientFn: groups.NewGroupMemberClient}),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube              client.Client
+	newGitlabClientFn func(cfg clients.Config) groups.GroupMemberClient
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.GroupMember)
+	if !ok {
+		return nil, errors.New(errNotGroupMember)
+	}
+	cfg, err := clients.GetConfig(ctx, c.kube, cr)
+	if err != nil {
+		return nil, err
+	}
+	return &external{kube: c.kube, client: c.newGitlabClientFn(*cfg)}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client groups.GroupMemberClient
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
```

Comment:
I believe the lateInitialize methods are missing. See for example: https://github.com/crossplane-contrib/provider-gitlab/blob/5029e6b21f939b2b632d900fbf69ed84be63df76/pkg/controller/groups/group.go#L217

Reply:
Maybe we can bump the [crossplane-runtime](https://github.com/crossplane/crossplane-runtime/releases/tag/v0.14.0) dependency and make use of the new feature by @muvaf for late initialization: https://github.com/crossplane/crossplane-runtime/pull/266

Reply:
> Maybe we can bump the [crossplane-runtime](https://github.com/crossplane/crossplane-runtime/releases/tag/v0.14.0) dependency and make use of the new feature by @muvaf for late initialization: [crossplane/crossplane-runtime#266](https://github.com/crossplane/crossplane-runtime/pull/266)

In this case, I see no reason to create func lateInitialize because in this case there are only two values, one of which is required to be filled, and the other can be nil

Reply:
It will be look like func lateInitialize(in *v1alpha1.GroupmemberParameters, group *gitlab.GroupMember) { 
	if groupMember == nil {
		return
	}
}

Reply:
Makes sense, thanks for explaining!